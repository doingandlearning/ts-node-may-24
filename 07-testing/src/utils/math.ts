export function add(a: number, b: number) {
  if (typeof a !== "number" || typeof b !== "number") {
    throw new Error("Both arguments should be numbers");
  }
  return a + b;
}
