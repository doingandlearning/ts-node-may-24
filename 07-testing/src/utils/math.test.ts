import { add } from "./math";

// describe - group tests together
// it/test

describe("happy path tests", () => {
  test("add two numbers together correctly", () => {
    // Arrange - Given
    const num1 = 7;
    const num2 = 10;
    const want = 17;

    // Act - When
    const result = add(num1, num2);
    // Assert - Then
    expect(result).toBe(17);
  });
});

test("this will fail if i don't pass in numbers", () => {
  const num1 = true;
  const num2 = "1";
  expect(() => add(num1 as any, num2 as any)).toThrow();
  expect(() => add(num1 as any, num2 as any)).toThrowErrorMatchingSnapshot();
});
