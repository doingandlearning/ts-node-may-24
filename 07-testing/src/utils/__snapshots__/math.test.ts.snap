// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`this will fail if i don't pass in numbers 1`] = `"Both arguments should be numbers"`;
